### PR TITLE
(enhancement) added ability to resolve aliases in meta data.

### PR DESCRIPTION
**Pull request by trazek**
Added ability to resolve aliases in meta data. 
Added test for resolving aliases in meta data.

### CODE REVIEW

File: lib/definition.js [hunk 1]
```diff
@@ -174,25 +181,57 @@ const transform = (def, options) => {
...
+
+  /**
+   * Resolve aliases inside prop meta data
+   */
+  const resolveMetaAliases = def => {
+    return def.update("props", data => {
+      const aliases = def.get("aliases", Immutable.Map());
+
+      return data.map((value, key) => {
+        if (Immutable.Map.isMap(value.get("meta"))) {
```
Comment (trazek) [inline lib/definition.js]: If there is no meta data, just pass through with original data. Do not attempt to resolve

File: lib/__tests__/definition.js [hunk 2]
```diff
@@ -261,6 +261,36 @@ it("resolves nested aliases", done => {
...
+    Immutable.Map({ includeMeta: true })
+  ).fold(
+    e => {
+      throw new Error(e);
+    },
+    r => {
+      const value = r.getIn(["props", "TOKEN_A", "value"]);
+      const metaValue = r.getIn(["props", "TOKEN_A", "meta", "meta_test"]);
+
+      expect(value).toEqual("#FF0000");
```
Comment (trazek) [inline lib/__tests__/definition.js]: Test to ensure original resolution works as well as meta data resolution

**Comment by trazek** [comment]
@aputinski @tomger @corygibbons

**Comment by bruno-gouveia** [comment]
I would really love this. I want to expose de baseFontPixel as a token and also use it as a meta property to export for ios. It would be great if both uses would have the same source of truth in an alias.

File: lib/definition.js [hunk 3]
```diff
@@ -43,6 +43,12 @@ const transform = (def, options) => {
             ? Either.of(def)
             : Either.try(resolveAliases)(def)
       )
+      .chain(
+        def =>
+          options.get("resolveMetaAlises") === false
```
Comment (aputinski) [inline lib/definition.js]:
```suggestion
          options.get("resolveMetaAliases") === false
```

**Comment by aputinski** [comment]
Thanks for the PR! This looks good to me.

**Comment by trazek** [comment]
> Thanks for the PR! This looks good to me. Just need to fix that one typo and then add the configuration option to the README.

Which configuration option are you referring about? An example of resolving aliases in metadata?

**Comment by aputinski** [comment]
Looks like you added the `resolveMetaAliases` option here:
https://github.com/salesforce-ux/theo/pull/173/files#diff-1c55b374199fe480050aa6172ce5f145R48

**Comment by aputinski** [comment]
We can add `resolveAliases` and `resolveMetaAliases` to the [API section](https://github.com/salesforce-ux/theo#api).

As far as version, what if we defaulted the option to `false` and then we can do a minor version release?

We should add this to the CHANGELOG as well.

**Comment by trazek** [comment]
> We can add `resolveAliases` and `resolveMetaAliases` to the [API section](https://github.com/salesforce-ux/theo#api).
> 
> As far as version, what if we defaulted the option to `false` and then we can do a minor version release?
> 
> We should add this to the CHANGELOG as well.

Defaulting `resolveMetaAliases` to false sounds good to me. I'll add the necessary information to the API section and CHANGELOG. Thanks

File: CLI.md [hunk 4]
```diff
@@ -12,39 +12,44 @@ $ theo <[file]> [options]
...
 A valid setup module exports a function that takes theo as the first argument.
 
 Example module (example.js):
+
 ```
 module.exports = theo => {
   theo.registerValueTransform(
-    'addpx', 
-    prop => prop.get('type') === 'size', 
+    'addpx',
```
Comment (trazek) [inline CLI.md]: Formatting changes are due to my VSCode prettier settings. Hopefully these changes do not violate any code standard.

File: bin/scripts/build.js [hunk 5]
```diff
@@ -5,7 +5,14 @@ const theo = require("../../lib");
...
 const path = require("path");
 
-module.exports = ({ src = "", dest, setup, formats, transform }) => {
+module.exports = ({
+  src = "",
+  dest,
+  setup,
+  formats,
+  transform,
+  resolveMetaAliases
```
Comment (trazek) [inline bin/scripts/build.js]: Passing `resolveMetaAliases` down in case the user sets it

File: lib/__tests__/definition.js [hunk 6]
```diff
@@ -291,6 +291,36 @@ it("resolves aliases in meta data", done => {
   );
 });
 
+it("DOES NOT resolve aliases in metadata when resolveMetaAliases flag is NOT set (false)", done => {
```
Comment (trazek) [inline lib/__tests__/definition.js]: Test to ensure not setting the `resolveMetaAliases` option, defaulted to `false` will not resolve aliases in metadata

File: bin/theo.js [hunk 7]
```diff
@@ -11,7 +11,8 @@ const options = {
   dest: argv.dest,
   setup: argv.setup,
   formats: (argv.format || "raw.json").split(","),
-  transform: argv.transform || "raw"
+  transform: argv.transform || "raw",
+  resolveMetaAliases: argv.resolveMetaAliases || false
```
Comment (trazek) [inline bin/theo.js]: Default `resolveMetaAliases` to `false` if not set

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "theo",
-  "version": "8.0.1",
+  "version": "8.1.1",
```
Comment (trazek) [inline package.json]: Bumping minor version, per PR conversation

**Comment by trazek** [comment]
Addressed PR comments @aputinski

**Comment by trazek** [comment]
Anything else you want me to add/change @aputinski?

**Comment by trazek** [comment]
Not a problem. Just wanted to make sure there wasn't anything else. Thanks!

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [8.1.1]
+
+### ⚠️ Breaking changes
```
Comment (aputinski) [inline CHANGELOG.md]: Can you remove this since it's no longer a breaking change? Once that's fixed we can merge / release. Thanks!

**Comment by trazek** [comment]
Will do

File: CHANGELOG.md (hunk 9 above)
Comment (trazek) [reply CHANGELOG.md]: Done. Removed the breaking changes notice.

**Comment by trazek** [comment]
Thanks for merging!

**Pull request merged**